fixed-hash/Cargo.toml

```diff
@@ -9,26 +9,30 @@ description = "Macros to define custom fixed-size hash types"
 documentation = "https://docs.rs/fixed-hash/"
 readme = "README.md"
 edition = "2021"
-rust-version = "1.60"
+rust-version = "1.64.0"
 
 [package.metadata.docs.rs]
-features = ["quickcheck", "api-dummy"]
+features = ["arbitrary", "quickcheck", "api-dummy"]
 
 [dependencies]
-quickcheck = { version = "1", optional = true }
-rand = { version = "0.8.0", optional = true, default-features = false }
-rustc-hex = { version = "2.0.1", optional = true, default-features = false }
+const-hex = { version = "1.13", default-features = false }
```

Comment:
why is `const-hex` not an optional dependency because `rustc-hex` was that before?

Reply:
if `rustc-hex` has a trait, like `rustc_hex::Trait`, and we need to impl `rustc_hex::Trait` for `H256`, then I agree that we should add `rustc-hex` as an optional feature.

Reply:
I'm more concerned with a larger dependency tree for users that can't opt-out from  it rather than a small impl inside this crate itself...

Lots of crates in this repo are very old and I don't think had a policy when to make it optional or not. Probably that's why we haven't been consistent

Anyway, I prefer to make `const-hex` an optional dependency such that users can opt-out from it but included in the default features.

Reply:
But the most users don't learn anything from the name of the `const-hex` feature, because it is used in the implementation of `std::str::FromStr`.
so why we add an optional feature for a std trait impl, unless the impl has multiple implementations based on different features.

Reply:
Ok, my cents just introduce a `feature hex` which brings enables `"dep:const-hex"` as an optional dependency and then do the same `uint` in this or another PR so we are consistent.

If we already are breaking things let's make it consistent...

@ordian any other ideas about this?

Reply:
To rephrase what I mean if the feature is not part of the core functionality of the crate/type then it should be an optional feature IMO (especially if it needs extra dependencies)

I don't follow why you mean with H256 because it's re-exported from several crates such as `fixed_hash, primitive_types and ethereum_types`

For instance for ethereum-types it makes sense to have `rustc-hex` as hard dependency because it's required to be able to parse hex strings in the ethereum-world IIRC but it's not necessarily case for fixed-hash because it's depends how it used but in most use-cases it's probably useful to have.

So, if you can motivate why it's essential `From::hex_str` for these types then I'm fine to have it has hard dependency but we shouldn't use different libraries in `uint` and `fixed_hash` for those. If `const_hex` is the way to go we should use it in `uint` as well i.e, not both use `hex and const_hex`

Reply:
> I don't follow why you mean with H256 because it's re-exported from several crates such as fixed_hash, primitive_types and ethereum_types

`H256` is just a example type.

> If const_hex is the way to go we should use it in uint as well i.e, not both use hex and const_hex

see description:
> BTW, I'm considering replacing both hex and rustc-hex in dependency tree with const-hex.

Reply:
OK, let me summarize:

1. Is `core::str::FromStr` one of the core functions of `fixed-hash`? IMO, it is, so I will choose hex as its str parsing format (it will rely on a hex parsing library, whether it is `const-hex`/`hex`/`rustc-hex`, I choose `const-hex`)
2. Why do I say that the existing feature declaration (like `rustc-hex` feature) is not a good idea? IMO, features are either used for different implementations of the same function, or used to support third-party libraries. However, from the existing design of `fixed-hash`, it should not support third-party libraries through features (I think it‘s to avoid the restrictions of the orphan rule), so there are specific types declared in `primitive-types` and `ethereum-types` and a series of `impls` crates.

Reply:
Okey, thanks for the explanation.

It's not necessarily requires to have a `FromStr` implementation to support parsing hex strings for fixed-hash but 
as you say it's quite annoying to have a feature-flag for it and makes it harder for everyone.

So, I'm onboard to make this `hex feature` required for the hash types but in the future if you want change stuff it's a much better way to open an issue to explain why something is bad and why you want to change it.

It's a more likely that we reject PRs that make breaking changes.